lib/src/source/git.dart

```diff
@@ -40,6 +41,49 @@ class GitSource extends CachedSource {
     });
   }
 
+  /// Gets the list of all versions from git tags.
+  /// Anything that Version.parse understands is considered a version,
+  /// it will also attempt to strip off a preceding 'v' e.g. v1.0.0
+  Future<List<Pubspec>> getVersions(String name, description) async {
+    if (!_useVersionTags(description)) {
```

Comment:
Yikes, good point. I'd just change `Source.getVersions()` to take a `PackageRef` rather than a name and description, and check whether the version constraint is `any` or not.